src/components/wb-playground/kyn-toast-wb/kyn-toast-wb.ts

```diff
@@ -0,0 +1,169 @@
+
+import closeIcon from '@carbon/icons/es/close/20';
+import successIcon from '@carbon/icons/es/checkmark--filled/16';
+import warningErrorIcon from '@carbon/icons/es/warning--alt--filled/16';
+import infoIcon from '@carbon/icons/es/information--filled/16';
+
+/**
+ * Toast notification.
+ * @slot anchor - Slot for the anchor button content.
+ */
+@customElement('kyn-toast-wb')
+export class KynToastWb extends LitElement {
+  static override styles = [ToastStyles];
+
+  /** Date formatting function
+   * @internal
+   */
+  private static _formatDate = _formatDate;
+
+  /**
+   * Body text, optional
+   * @type {boolean}
+   * @default false
+   */
+  @property({ type: Boolean })
+  showToast = false;
+
+  /**
+   * Toaster notification mode/status. `'default'`, `'success'`, `'warning'`, or `'error'`.
+   * @type {string}
+   * @default 'success'
+   */
+  @property({ type: String })
+  toastStatus = 'default';
+
+  /**
+   * Title/heading text, required.
+   * @type {string}
+   * @default ''
+   */
+  @property({ type: String })
+  toasterTitle = '';
+
+  /**
+   * Body text, optional
+   * @type {string}
+   * @default ''
+   */
+  @property({ type: String })
+  toasterBody = '';
+
+  /**
+   * Duration that toaster is visible to the user
+   * @type {number}
+   * @default 5000
+   */
+  @property({ type: Number })
+  toastLifespan = 5000;
+
+  /** internal timestamp property
+   * @internal
+   */
+  @state()
+  _timestamp = KynToastWb._formatDate(Date.now());
+
+  /**
+   *Timestamp visible state
+   * @type {boolean}
+   * @default false
+   */
+  @property({ type: Boolean })
+  timestampVisible = false;
+
+  /** Toaster element
+   * @internal
+   */
+  @query('#toaster')
+  _toaster!: HTMLDivElement;
+
+  /** Interval to hide toaster
+   * @internal
+   */
+  private _toastTimeoutId: number | undefined = undefined;
+
+  override render() {
+    const notificationIcon: any = {
+      success: successIcon,
+      error: warningErrorIcon,
+      warning: warningErrorIcon,
+      default: infoIcon,
+    };
+
+    return html`
+      <span
+        class="anchor"
+        @click=${() => this._toggleToast(this.toastLifespan)}
+      >
+        <slot name="anchor"></slot>
+      </span>
+      <div
+        id="toaster"
+        class=${`${this.toastStatus} toast-visible-${this.showToast}`}
+        ?hidden=${!this.showToast}
+      >
+        <button class="x-out" @click=${this._toggleToast}>
+          <kd-icon .icon=${closeIcon}></kd-icon>
+        </button>
+        <p id="toaster-title" class=${this.toastStatus}>
+          <kd-icon .icon=${notificationIcon[this.toastStatus]}></kd-icon>${this
+            .toasterTitle}
+        </p>
+        <p id="toaster-body">${this.toasterBody}</p>
+        ${this.timestampVisible
+          ? html`<span id="timestamp">${this._timestamp}</span>`
+          : null}
+      </div>
+    `;
+  }
+
+  override updated(changedProperties: PropertyValues<this>): void {
```

Comment:
I planned to talk a bit a bout Lit Lifecycle on Friday, but just a note on `updated` vs `willUpdate`. `updated` fires after render, so if you modify any reactive properties at that time it will trigger another update cycle. `willUpdate` triggers before the render, which allows you to make changes beforehand without triggering a re-render.